ui/doc/developer-setup.md

```diff
@@ -0,0 +1,35 @@
+# Developer Setup
+
+## Prequisties
+
+- Linux or MacOS
+- [VS Code](https://code.visualstudio.com/download)
+- [Docker Desktop](https://www.docker.com/products/docker-desktop/)
+- [NPM](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
+
+## Setup
+
+- Clone the repo locally and open in VS Code
+- Open a terminal and navigate to the `ui` folder
+
+## Start the UI
+- Run `make` to run the `./Makefile` installation script
+    - On the "Watch Usage" prompt, press `q` to continue the script
+- Run `npm start` 
+    - This will start the front-end and open it in the browser
+- Press `login` to continue in your browser
+- Press `CTRL + C` in the terminal window to quit the npm start
+- Run ` ./run.py &` to standup a mock API request from Azure
+
+## Start the server
+
```

Comment:
We should a disclaimer here that running server actually runs the deployment engine and it requires several environment variables in order to connect to Azure.
Also, it is missing step `cd server`.